dove/tests/test_cmd_helper.rs

```diff
@@ -0,0 +1,143 @@
+#![cfg(test)]
+#![allow(dead_code)]
+
+use std::path::{PathBuf, Path};
+use std::fs::{remove_dir_all, read_to_string, create_dir_all};
+use dove::cli::execute;
+use fs_extra::file::write_all;
+use toml::Value;
+
+pub fn project_start(project_name: &str) -> (PathBuf, PathBuf) {
+    let tmp_folder = std::env::temp_dir();
+    let project_folder = tmp_folder.join(project_name);
+    if project_folder.exists() {
+        remove_dir_all(&project_folder).unwrap_or_else(|_| {
+            panic!(
+                "[ERROR] Couldn't delete project directory: {}",
+                project_folder.display()
+            )
+        });
+    }
+    (tmp_folder.clone(), project_folder)
+}
+pub fn project_start_for_init(project_name: &str) -> PathBuf {
+    let (_, project_folder) = project_start(project_name);
+    project_remove(&project_folder);
+
+    // Create project directory
+    create_dir_all(&project_folder).unwrap_or_else(|_| {
+        panic!(
+            "Failed to create directory: {}",
+            project_folder.to_str().unwrap_or(" - "),
+        )
+    });
+    project_folder
+}
+pub fn project_start_nb(project_name: &str) -> PathBuf {
+    let (base_folder, project_folder) = project_start(project_name);
+    project_new_default(&base_folder, &project_folder, project_name);
+    project_build(&project_folder);
+    project_folder
+}
+/// $ dove new ###
+pub fn project_new_default(base_folder: &Path, project_folder: &Path, project_name: &str) {
+    let args = &["dove", "new", project_name];
+    let command_string: String = args.join(" ").to_string();
+    execute(args, base_folder.to_path_buf()).unwrap_or_else(|err| {
+        panic!(
+            "[COMMAND] {}\r\n[FOLDER] {}\r\n[ERROR] {}\r\n",
+            &command_string,
+            base_folder.display(),
+            err.to_string()
+        )
+    });
+    set_dependencies_local_move_stdlib(&project_folder);
+}
+/// $ dove new ### -d ### -a ### -r ###
+pub fn project_new_with_args(
+    base_folder: &Path,
+    project_folder: &Path,
+    project_name: &str,
+    project_dialect: &str,
+    project_address: &str,
+    project_blockchain_api: &str,
+) {
+    let args = &[
+        "dove",
+        "new",
+        project_name,
+        "-d",
+        project_dialect,
+        "-r",
+        project_blockchain_api,
+        "-a",
+        project_address,
+    ];
+    let command_string: String = args.join(" ").to_string();
```

Comment:
I think you should extract this and the next line as smth like
```
fn execute_dove_at(project_url: PathBuf, args: &[str]):
```
so that following `panic!` call would only be written once. 

You can also add `setup_move_stdlib` boolean argument to this function which would eliminate most of the `set_dependencies_local_move_stdlib` calls after you move to invoking this function everywhere.